File: dogbin/lib/model/document.py
class Document:
    slug:str
    isUrl:bool
    content:str
    viewCount:int
    version:int
    owner:object

    def __init__(self, slug:str, isUrl:bool, content:str, viewCount:int=0, version:int=0, owner:object=None):
        self.slug = slug
        self.isUrl = isUrl
        self.content = content
        self.viewCount = viewCount
        self.version = version
        self.owner = owner

    def increaseViewCount(self):
        self.viewCount += 1

    def update_content(self, content:str):
        self.content = content
        self.version += 1

    def userCanEdit(self, user):
        return False

File: data/model/src/dog/del/data/model/Document.kt
package dog.del.data.model

import dog.del.commons.Date
import java.io.InputStream

interface Document<T: DocumentType, U: User<*>> {

    var slug: String

    var type: T

    var stringContent: String?
    var blobContent: InputStream?
    var version: Int

    var owner: U

    val created: Date
}

interface DocumentType